games-vue-client/src/components/chat/Chat.vue

```diff
@@ -0,0 +1,183 @@
+      :key="componentKey"
+      width="256"
+      class="chat"
+    >
+      <v-chip-group
+        active-class="active-chat"
+        column
+      >
+        <v-chip
+          v-for="chat in chats"
+          :key="chat.chatId"
+          close
+          close-icon="mdi-close"
+          @click:close="leaveChat(chat.chatId)"
+          @click="activeChat(chat.chatId)"
+        >
+          {{ chat.chatId }}
+        </v-chip>
+      </v-chip-group>
+      
+      <v-divider />
+
+      <v-card-title>
+        <span class="title font-weight-light">Chat {{ currentChat.chatId }}</span>
+
+        <v-col class="text-right">
+          <v-icon @click="toggleDisplay">
+            mdi-close
+          </v-icon>
+        </v-col>
+      </v-card-title>
+
+      <v-card-text>
+        <div class="messages">
+          <div
+            ref="container"
+            class="messages-body"
+          >
+            <div
+              v-for="message in currentChat.messages"
+              :key="message.message + message.from + message.date"
+            >
+              <span class="from">{{ message.from }}</span>
+              <span class="message">
+                {{ message.message }}
+              </span>
+            </div>
+          </div>
+        </div>
+
+        <v-textarea
+          v-model="newMessage"
+          label="Message"
+          rows="2"
+          counter
+          maxlength="100"
+          dense
+          full-width
+          autofocus
+          no-resize
+          hint="Please be respectful"
+          append-outer-icon="mdi-send"
+          @click:append-outer="sendMessage"
+          @keyup.enter="sendMessage"
+        />
+      </v-card-text>
+    </v-card>
+    <v-btn
+      v-else
+      elevation="2"
+      fab
+      class="open-chat"
+      @click="toggleDisplay"
+    >
+      Chat
+    </v-btn>
+  </div>
+</template>
+<script>
+export default {
+  name: "Chat",
+  props: [],
+  data() {
+    return {
+      display: true,
+      currentChat: {},
+      newMessage: '',
+      messages: [],
+      componentKey: 0,
+    };
+  },
+  methods: {
+    // forceRerender() {
+    //   this.componentKey += 1;
+    // },
+    scrollToEnd () {
+      var content = this.$refs.container;
+      content.scrollTop = content.scrollHeight;
+    },
+    toggleDisplay() {
+      this.display = !this.display;
+    },
+    sendMessage() {
+      if (this.newMessage.trim().length > 0) {
+        this.$store.dispatch("chat/message", { chat: { chatId: this.currentChat.chatId, message: this.newMessage.trim() } });
+        this.scrollToEnd();
+      }
+      this.newMessage = '';
+    },
+    activeChat(chatId) {
+      this.currentChat = this.chats[chatId];
+    },
+    leaveChat(chatId) {
+      this.$store.dispatch("chat/leave", { chat: { chatId } });
+    }
+  },
+  mounted() {
+    this.currentChat = this.chats.server;
+    this.scrollToEnd();
+  },
+  computed: {
+    chats() {
+      return this.$store.state.chat.chats;
```

Comment:
Using `this.$store.state` is not correct. See for example `InviteScreen.vue` for an example of how to use `mapState` to make reactivity work correctly

Reply:
Perfect!